mars/services/task/supervisor/preprocessor.py

```diff
@@ -203,6 +205,11 @@ def tile(self, tileable_graph: TileableGraph) -> Iterable[ChunkGraph]:
                     optimize_chunk_graph(chunk_graph)
                 )
             yield chunk_graph
+        # Extract ids of tileables whose op need func cleanup.
+        for t in tileable_graph:
+            if hasattr(t.op, "func_key") and t.op.func_key is not None:
+                assert t.op.need_clean_up_func is True
+                self.tileable_ids_need_clean_up_func.append(t.key)
```

Comment:
Is the `func_key` changed during execution? If not, append the `func_key` to `self.tileable_ids_need_clean_up_func` is better.